lib/browser/api/net.js

```diff
@@ -41,7 +41,46 @@ class IncomingMessage extends Readable {
   }
 
   get headers () {
-    return this.urlRequest.rawResponseHeaders
+    // list of headers that Node.js discards duplicates for
+    // see https://nodejs.org/api/http.html#http_message_headers
+    const discardableDuplicateHeaders = [
+      'content-type',
+      'content-length',
+      'user-agent',
+      'referer',
+      'host',
+      'authorization',
+      'proxy-authorization',
+      'if-modified-since',
+      'if-unmodified-since',
+      'from',
+      'location',
+      'max-forwards',
+      'retry-after',
+      'etag',
+      'last-modified',
+      'server',
+      'age',
+      'expires'
+    ]
+
+    const filteredHeaders = {}
+    const rawHeaders = this.urlRequest.rawResponseHeaders
+    Object.keys(rawHeaders).forEach(header => {
+      if (header in filteredHeaders && discardableDuplicateHeaders.includes(header)) {
+        // do nothing with discardable duplicate headers
+      } else {
+        if (header === 'set-cookie') {
+          // keep set-cookie as an array per Node.js rules
+          // see https://nodejs.org/api/http.html#http_message_headers
+          filteredHeaders[header] = rawHeaders[header]
+        } else {
+          // a header is present iff it has a value, and there will only be 1 value
```

Comment:
This doesn't seem right, the node.js docs say:
> For all other headers, the values are joined together with ', '.

Reply:
(resolved offline)